src/new/wallet.js

```diff
@@ -927,19 +940,19 @@ class HathorWallet extends EventEmitter {
    * @memberof HathorWallet
    * @inner
    **/
-  getPreProcessedData(key) {
+  async getPreProcessedData(key) {
     if (Object.keys(this.preProcessedData).length === 0) {
-      this.preProcessWalletData();
+      await this.preProcessWalletData();
     }
     return this.preProcessedData[key];
   }
 
-  setState(state) {
+  async setState(state) {
```

Comment:
We have this method in the new facade as well and it's not `async`. Maybe we should keep this method not async and just call `this.preProcessWalletData();` without await. It should have the same behaviour and we keep the same method signatures in both facades.

Reply:
I was worried about potential race conditions of setting the state to `READY` without the `preProcessWalletData`

But since the `getTxbalance` resolves immediately, we should be fine, I've tested it and noticed no issues so far.

Added a comment with a notice on d192e3d2df1485dc1641a742f3e2d8627487de1b

What do you think?

Reply:
I like it.